### ISSUE TITLE
LO plugin: ReferenceHeaderParagraphFormat messed up after 'refresh'

### ISSUE THREAD

**Issue by jdpipe** [JabRef/jabref#1646]
Often when I click the 'refresh' button in the LO toolbar, my 'References' heading above my bibliography reverts from being properly styled (all caps, bold) to unstyled (roman, no longer all caps). It still appears centred. The line still has the correct paragraph style, but for some reason the character style is temporarily lost. If I type 'ctrl-M' to remove direct formatting from that heading-paragraph, then it goes back to normal. But then if I click 'refresh' in Jabref again, it gets reverted to non-bold, non-allcaps again.

Looks like some mysterious code is being placed in the References heading when it is being 'typed' by Jabref, but I couldn't speculate how or why or where.

Tested with JabRef 3.6dev--snapshot--2016-07-29--extractoobibdatabase--fba22d9
linux 4.4.0-31-generic amd64 
Java 1.8.0_91. Also see with version 2.10 (Ubuntu 16.04 latest released version)

**Comment by oscargus** [comment]
Is this a new bug or did it happen before the database extraction was added?

**Comment by jdpipe** [comment]
I has been like this for a very long time. At least since 2012, I think.

**Comment by oscargus** [comment]
OK, great! (As in, "Great, I didn't cause it"...) Thanks for the feedback.

**Comment by oscargus** [comment]
I checked it now and it is not obvious what to do about it. The code is very straightforward, so I cannot really tell what might go wrong:

``` Java
text.insertString(cursor, string, true);
XParagraphCursor parCursor = UnoRuntime.queryInterface(XParagraphCursor.class, cursor);
 // Access the property set of the cursor, and set the currently selected text
 // (which is the string we just inserted) 
 XPropertySet props = UnoRuntime.queryInterface(XPropertySet.class, parCursor);
 props.setPropertyValue(PARA_STYLE_NAME, parStyle);
 ```
So basically, insert text, change the paragraph format of the currently selected/inserted text. I tried to step debug this and cannot really see what can be improved here.

However, I saw an issue/comment (that I cannot find at the moment), where the character format was changed a few characters away from where the user inserted references. This leads me to believe that the issues are somehow related and maybe there is something wrong with the OpenOffice/LibreOffice interface (or we are accessing it incorrectly, although it works most of the time).
